src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,207 @@
+package duke;
+
+import java.util.Scanner;
+
+import duke.exceptions.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+import duke.ui.Ui;
+import duke.util.TaskList;
+
+
+/**
+ * Duke is a personal assistant chatbot that helps a person to keep track of various things.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Constructs a Duke object.
+     */
+    public Duke() {
+        try {
+            this.ui = new Ui();
+            this.storage = new Storage();
+            this.tasks = new TaskList(storage.readTasks());
+            this.parser = new Parser();
+
+        } catch (DukeException e) {
+            ui.printErrorMessage(e);
+        }
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
```

Comment:
Perhaps you can update the header comment to reflect the description of the method better.

Reply:
Right, thanks for the suggestion, looks like my code needs some cleanup lol